### PR TITLE
Add `MatQ::simplify`

### PR DESCRIPTION
**Pull request by jnsiemer**
**Description**

This PR implements...
- [x] simplify

for `MatQ`.
This function might be used for matrices with large denominators before multiplying with another matrix to boost the efficiency of matrix multiplication. The current implementation isn't focused on efficiency and merely a check how much it might increase efficiency for matrix multiplication. All relevant tests are part of [Q::simplify].

**Testing**
- [x] I added basic working examples (possibly in doc-comment)
- [x] I added tests for large (pointer representation) values
- [x] I triggered all possible errors in my test in every possible way
- [x] I included tests for all reasonable edge cases

**Checklist:**
- [x] I have performed a self-review of my own code
  - [x] The code provides good readability and maintainability s.t. it fulfills best practices like talking code, modularity, ...
    - [x] The chosen implementation is not more complex than it has to be
  - [x] My code should work as intended and no side effects occur (e.g. memory leaks)
  - [x] The doc comments fit our style guide

### CODE REVIEW

File: src/rational/mat_q/rounding.rs [hunk 1]
```diff
@@ -95,6 +95,79 @@ impl MatQ {
...
+    /// Returns a matrix, where each entry was simplified using [`Q::simplify`],
+    /// i.e. each entry becomes the smallest rational with the smallest denominator in the range
+    /// `\[entry - |precision|, entry + |precision|\]`.
+    ///
+    /// This function allows to free memory in exchange for the specified loss of
+    /// precision (see Example 3). Be aware that this loss of precision is propagated by
+    /// arithmetic operations and can be significantly increased depending on the
+    /// performed operations.
+    ///
+    /// This function ensures that there is no sign change in any of the entries.
```
Comment (Marvin-Beckmann) [inline src/rational/mat_q/rounding.rs:107]: I do not understand this comment

**Review by Marvin-Beckmann — APPROVED** [review]
make sure to add the task of simpliying PolyOverQ to the kanboard.
(This is much more difficult due to the structure of how the PolyOverQ are saved for PolyOverQ, as the naive approach would not work there),
